### PR TITLE
Enable copy-on-write heap initialization by default

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit enables the `Config::memfd` feature by default now that it's
been fuzzed for a few weeks on oss-fuzz, and will continue to be fuzzed
leading up to the next release of Wasmtime in early March. The
documentation of the `Config` option has been updated as well as adding
a CLI flag to disable the feature.

### CODE REVIEW

File: crates/wasmtime/src/config.rs [hunk 1]
```diff
@@ -1172,26 +1172,45 @@ impl Config {
...
-    /// If all of the above applies, this setting is enabled, and the current
-    /// platform is Linux the `memfd` will be used to efficiently initialize
-    /// linear memories with `mmap` to avoid copying data from initializers into
-    /// linear memory.
+    /// Modules which do not meet these criteria will fall back to
+    /// initialization of linear memory based on copying memory.
+    ///
+    /// This feature of Wasmtime is also platform-specific:
+    ///
+    /// * Unix - this feature is only supported when loading modules from a
```
Comment (cfallin) [inline crates/wasmtime/src/config.rs:1198]: maybe we can put "Unix" after "Linux" in this list and say "Other Unix" instead, for clarity?

**Review by cfallin — APPROVED** [review]
Thanks!

A random thought occurs: now that this feature has been generalized to work with generic Unix and mmap files straight from disk, with no actual `memfd_create()`, I wonder if there's a better name for it, like "copy-on-write heap images"? (`cow_heaps`?) To be clear I don't think it absolutely needs to change; it's well-documented exactly what it means in any case; but it might avoid some confusion at the margins in the future if it doesn't refer to the Linux-specific feature.

**Comment by alexcrichton** [comment]
I went ahead and merged https://github.com/bytecodealliance/wasmtime/pull/3831 because I read that first and rebased here to update the names/docs w.r.t. "memfd" terminology

File: crates/wasmtime/src/config.rs (hunk 1 above)
Comment (Gam3356) [reply crates/wasmtime/src/config.rs]: Thank you guys! Willie Parish 12 Pro max ![Uploading 173412302...]

**Pull request merged**